### DISCUSSION TITLE
Can't add/drag components to canvas of a blank screen

### DISCUSSION THREAD

**Discussion by neshorg** [Budibase/budibase#14521] [Help] [answered]
Hello, 

I'm new to Budibase and just created a new app to test a few things. 
After creating a new blank screen, I can't seem to add new components to the screen. 

This seems very basic and I feel like I'm missing something very simple perhaps. Can anyone help me out?

P.S.: Btw, I've created an app from a template and tried the same thing by creating a new blank screen to that app to try and create my own custom screen. However, I got the same results there as well. 

docker version 2.31.1
EDIT: I've upgraded to 2.31.6, but still same results

![blankScreen_componentIssue](https://github.com/user-attachments/assets/827eff4c-2759-43a9-926d-9b44539cfdb2)

**Comment by ConorWebb96** [comment, marked as answer]
Hi @neshorg,

The new grid layout doesn't support drag and drop of components from the builder. You can click the components to add them to the screen and then drag/move/resize them on your screen.

If you add a container to your page for either layout you can drag the components into it. I hope this helps.

If you wish to go back to the old layout you can toggle the setting in the screens area, its called layout. I've attached a screenshot of this below.
<img width="326" alt="Screenshot 2024-09-05 at 12 50 00" src="https://github.com/user-attachments/assets/54e6d42b-e06b-42c0-9de3-4136d40918c0">


We will be looking into adding the old drag and drop functionality back into the builder. 

Thank you for your feedback!

**Reply by neshorg** [reply]
Thanks @ConorWebb96,

I can't believe I missed this. I knew it had to be something simple, but never thought of clicking on the component. 

Adding components by clicking on them from the drawer is fine for me by the way. 
Dragging them on the canvas would be nice, but not particularly necessary as long as you can drag them afterwards as desired. From what I understand, this is possible by switching the layout in order to drag components right?

**Reply by ConorWebb96** [reply]
I'm happy to have helped @neshorg 

I believe you will still need to add a component initially before it starts to work but yes, it will work for the old layout option. A good first component I always start with is the container.

Thanks again for the feedback!
